### ISSUE TITLE
GPU not used on Apple Silicon (M1 Pro) resulting in very slow inference

### ISSUE THREAD

**Issue by sammcj** [KillianLucas/open-interpreter#721]
### Describe the bug

I've followed the instructions for installing OI on macOS however when inference is running the GPU is not utilised at all, there is a single Python process using between 150-350% CPU.

Interference is incredibly slow with Mistral 7b q4, taking several minutes to answer simple queries such as "list all files in the current directory".

Note below that it seems OI is not using the selected Python version and instead seems to be using the older OS Python version - maybe this is related?

### Reproduce

1. Use a M1 Pro mac (running Sonoma)
2. Follow https://github.com/KillianLucas/open-interpreter/blob/main/docs/MACOS.md to install OI
3. Run OI with `--local` and select `mistral-7b-instruct-v0.1.Q4_0.gguf`
4. Input something like "list all files in the current directory"
5. Note that the GPU is not utilised

### Expected behavior

When using Ollama or LM Studio inferences is very fast, especially on small models like Mistral 7b, I'd assume that OI should perform similarly although perhaps slightly slower due to an increased context size (maybe?).

### Screenshots

<img width="765" alt="image" src="https://github.com/KillianLucas/open-interpreter/assets/862951/0796c37b-88bc-44bb-b45c-91d9335151f4">

### Open Interpreter version

0.1.10

### Python version

3.11.3

Note that in the above screenshot you can see Python 3.10, I'm guessing OI must specify a specific version when launched?

- 3.11.3 is what is returned from `python -V`
- 3.11.3 is shown with `pip info llama-cpp-python`

```plain
pip show llama-cpp-python                                                                              Name: llama_cpp_python
Version: 0.2.11
Summary: Python bindings for the llama.cpp library
Home-page:
Author:
Author-email: Andrei Betlen <abetlen@gmail.com>
License: MIT
Location: /Users/samm/.pyenv/versions/3.11.3/lib/python3.11/site-packages
Requires: diskcache, numpy, typing-extensions
Required-by:
```

### Operating System name and version

macOS 14.1

### Additional context

```
pip list|grep -Ei 'torch|formers|cv|llama|accel'
accelerate                                        0.23.0
llama_cpp_python                                  0.2.11
opencv-contrib-python                             4.8.1.78
opencv-python                                     4.8.1.78
opencv-python-headless                            4.7.0.72
sentence-transformers                             2.2.2
torch                                             2.0.1
torchvision                                       0.15.2
transformers                                      4.29.0
```

**Comment by ponnhide** [comment]
@sammcj 
I encountered the same issue. It appears that the current open-interpreter creates another conda environment with Python 3.10 in '/Users/{YOUR_NAME}/.ooba/text-generation-ui/installer_files/env/bin/' and runs the LLM under this environment. This likely explains why the Mac's GPU is currently not being utilized. 
I downgraded the open-interpreter to version 0.1.4 and confirmed that the Mac can now properly use its GPU
